Doc/whatsnew/3.7.rst

```diff
@@ -1012,90 +1916,193 @@ Other CPython Implementation Changes
-  been used. If the specified version is not available py.exe will error exit.
-  (Contributed by Steve Barnes in :issue:`30291`.)
+importlib
+---------
+
+Methods
+:meth:`MetaPathFinder.find_module() <importlib.abc.MetaPathFinder.find_module>`
+(replaced by
+:meth:`MetaPathFinder.find_spec() <importlib.abc.MetaPathFinder.find_spec>`)
+and
+:meth:`PathEntryFinder.find_loader() <importlib.abc.PathEntryFinder.find_loader>`
+(replaced by
+:meth:`PathEntryFinder.find_spec() <importlib.abc.PathEntryFinder.find_spec>`)
+both deprecated in Python 3.4 now emit :exc:`DeprecationWarning`.
+(Contributed by Matthias Bussonnier in :issue:`29576`)
+
+The :class:`importlib.abc.ResourceLoader` ABC has been deprecated in
+favour of :class:`importlib.abc.ResourceReader`.
+
+
+locale
+------
+
+:func:`locale.format` has been deprecated, use :meth:`locale.format_string`
+instead.  (Contributed by Garvit in :issue:`10379`.)
+
+
+macpath
+-------
+
+The :mod:`macpath` is now deprecated and will be removed in Python 3.8.
+
+
+threading
+---------
+
+:mod:`dummy_threading` and :mod:`_dummy_thread` have been deprecated.  It is
+no longer possible to build Python with threading disabled.
+Use :mod:`threading` instead.
+(Contributed by Antoine Pitrou in :issue:`31370`.)
+
+
+socket
+------
+
+The silent argument value trunctation in :func:`socket.htons` and
+:func:`socket.ntohs` has been deprecated.  In future versions of Python,
+if the passed argument is larger than 16 bits, an exception will be raised.
+(Contributed by Oren Milman in :issue:`28332`.)
+
+
+ssl
+---
+
+:func:`ssl.wrap_socket` is deprecated.  Use
+:meth:`ssl.SSLContext.wrap_socket` instead.
+(Contributed by Christian Heimes in :issue:`28124`.)
+
+
+sunau
+-----
+
+:func:`sunau.openfp` has been deprecated and will be removed in Python 3.9.
+Use :func:`sunau.open` instead.
+(Contributed by Brian Curtin in :issue:`31985`.)
+
+
+sys
+---
+
+Deprecated :func:`sys.set_coroutine_wrapper` and
+:func:`sys.get_coroutine_wrapper`.
+
+The undocumented ``sys.callstats()`` function has been deprecated and
+will be removed in a future Python version.
+(Contributed by Victor Stinner in :issue:`28799`.)
+
+
+wave
+----
+
+:func:`wave.openfp` has been deprecated and will be removed in Python 3.9.
+Use :func:`wave.open` instead.
+(Contributed by Brian Curtin in :issue:`31985`.)
 
-- The launcher can be run as ``py -0`` to produce a list of the installed pythons,
-  *with default marked with an asterisk*. Running ``py -0p`` will include the paths.
-  If py is run with a version specifier that cannot be matched it will also print
-  the *short form* list of available specifiers.
-  (Contributed by Steve Barnes in :issue:`30362`.)
 
+Deprecated functions and types of the C API
+===========================================
+
+Function :c:func:`PySlice_GetIndicesEx` is deprecated and replaced with
+a macro if ``Py_LIMITED_API`` is not set or set to the value between
```

Comment:
... is not set, is set to a value in the range between ``0x03050400`` and ``0x03060000`` (not inclusive), or is ``0x03060100`` or higher.